src/rt/cpp/when.h

```diff
@@ -15,6 +15,12 @@ namespace verona::cpp
 {
   using namespace verona::rt;
 
+#define BEHAVIOUR_YIELD(X) \

```

Comment:
Could you move this Macro to the test.  I don't really want C macros in the C++ header files.